.changeset/giant-nails-grow.md

```diff
@@ -0,0 +1,16 @@
+---
+'@backstage/plugin-auth-backend': minor
+---
+
+- Moved `IdentityClient`, `BackstageSignInResult`, `BackstageIdentityResponse`,
+  and `BackstageUserIdentity` to `@backstage/plugin-auth-node`.
```

Comment:
Let's call out all the breaking changes with `**BREAKING**:`. Feeling particularly this one needs more attention as it affects custom auth providers. Potentially even import and re-export there for a period of deprecation

Reply:
Yeah good call, added a diff to help users migrate too